forge/db/controllers/Device.js

```diff
@@ -17,7 +17,7 @@ module.exports = {
             // Check the snapshot is one we recognise
             const snapshotId = app.db.models.ProjectSnapshot.decodeHashid(state.snapshot)
             // hashid.decode returns an array of values, not the raw value.
-            if (snapshotId.length > 0) {
+            if (snapshotId?.length > 0) {
```

Comment:
If `state.snapshot` was not found in the DB this was raising an error (that was previously silently dropped)